File: TCCC2001Round2/Football.py
# Problem Statement Link: https://community.topcoder.com/stat?c=problem_statement&pm=42&rd=2001

class Football():
    def __init__( self ):
        self.__scores = []
        self.__scores.append( 2 )
        self.__scores.append( 3 )
        self.__scores.append( 7 )

    def fetchCombinations( self, n ):
        dp = [ [ 0 for i in range( len( self.__scores ) ) ] for i in range( n + 1 ) ]
        for i in range( n + 1 ):
            for j in range( len( self.__scores ) ):
                if not i:
                    dp[ i ][ j ] = 1
                else:
                    if j > 0:
                        dp[ i ][ j ] = dp[ i ][ j - 1 ]
                    if i - self.__scores[ j ] >= 0:
                        dp[ i ][ j ] += dp[ i - self.__scores[ j ] ][ j ]
        return dp[ n ][ len( self.__scores ) - 1 ]


n = int( raw_input() )
football = Football()
print football.fetchCombinations( n )

File: TCCC2001Round2/Football.js
function Football() {
    this.__scores = [ 2, 3, 7 ];
}

Football.prototype.fetchCombinations = function( n ) {
    var dp = [];
    for( var i = 0; i <= n; ++i ) {
        dp.push( 0 );
    }
    dp[ 0 ] = 1;
    for( var i = 0; i < this.__scores.length; ++i ) {
        for( var j = this.__scores[ i ]; j <= n; ++j ) {
            dp[ j ] += dp[ j - this.__scores[ i ] ];
        }
    }
    return dp[ n ];
}

var football = new Football();
for( var n = 0; n < 76; ++n ) {
    console.log( "n = ", n );
    console.log( football.fetchCombinations( n ) );
}
